internal/gensupport/send_test.go

```diff
@@ -31,6 +34,39 @@ func TestSendRequestWithRetry(t *testing.T) {
 	}
 }
 
+type headerRoundTripper struct {
+	wantHeader http.Header
+}
+
+func (rt *headerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	// Ignore x-goog headers sent by SendRequestWithRetry
+	delete(r.Header, "X-Goog-Api-Client")
+	delete(r.Header, "X-Goog-Gcs-Idempotency-Token")
```

Comment:
nit: there is a [`Header.Del`](https://pkg.go.dev/net/http#Header.Add) which is case insensitive - probably good to use.